### DISCUSSION TITLE
Color coding for `GD.Print(...)`

### DISCUSSION THREAD

**Discussion by valkyrienyanko** [godotengine/godot-proposals#5756] [Editor]
Being able to color code certain messages when using `GD.Print(...)` would allow me in my case to color code all client messages one color and all server messages another color making it easier to see what's being logged from what in the Godot editor console.

![image](https://user-images.githubusercontent.com/6277739/200943958-c2633e21-2592-41fa-8586-2391d2555f5e.png)

**Comment by Calinou** [comment]
This is already available in 4.0 as `print_rich()` (`GD.PrintRich()` in C#). It can't be backported to 3.x though as it relies on many low-level changes.

In the meantime, what you can do is prepend a few tabs of indentation (`\t`) to your server messages to distinguish them from client messages.

**Reply by valkyrienyanko** [reply]
Oh sorry I should have looked harder before opening up a proposal about this. Thanks!

**Comment by awardell** [comment]
You may be interested in this plugin https://github.com/4d49/godot-logger/tree/3.x

I haven't used it and so I cannot vouch for it, but it looks like it should be able to provide you with a logger with color options in 3.X

print_rich is working great in 4.X
